core/dbt/parser/unit_tests.py

```diff
@@ -59,14 +61,25 @@ def parse_unit_test_case(self, test_case: UnitTestDefinition):
         # for selection.
         # Note: no depends_on, that's added later using input nodes
         name = f"{test_case.model}__{test_case.name}"
+        if test_case.expect.format == "dict":
+            if isinstance(test_case.expect.rows, List):
+                expected_rows = test_case.expect.rows
+            else:
+                raise ParsingError("Wrong format for expected rows")
+        else:  # test_case.expect.format == "csv":
+            # build a dictionary from the csv string
+            if isinstance(test_case.expect.rows, str):
+                expected_rows = self._build_rows_from_csv(test_case.expect.rows)
+            else:
+                raise ParsingError("Wrong format for expected rows")
```

Comment:
I'm brainstorming ways that we could consolidate this logic and share it across `InputFixture` and `OutputFixture`. The simplest thing I can think of is adding a `parsed_rows` property on a shared `UnitTestFixture` base class that does this conditional logic to parse the raw `rows` given a `format` and convert it into a consistent internal representation. What do you think?

Reply:
An alternative could be to have a `UnitTestFixture` class that can be created from either an `InputFixture` or `OutputFixture` (making these just the external-facing interfaces). That way we could encapsulate the logic of parsing the rows based on the format, into a consistent internal fixture object and store the parsed rows as state on that object.

Reply:
Yeah, I thought it would be a good idea to have some kind of combined logic but when I wrote this my brain was kind of offline :).   Since we know we're going to also support csv files soon, it should be a solution that makes that easy.  Let me think about whether a base class or derived class makes more sense...